mypyc/lib-rt/bytes_ops.c

```diff
@@ -95,15 +95,26 @@ PyObject *CPyBytes_GetSlice(PyObject *obj, CPyTagged start, CPyTagged end) {
     return CPyObject_GetSlice(obj, start, end);
 }
 
-// Like _PyBytes_Join but fallback to dynamic call if 'sep' is not bytes
-// (mostly commonly, for bytearrays)
 PyObject *CPyBytes_Join(PyObject *sep, PyObject *iter) {
+#if CPY_3_13_FEATURES
+    PyObject *args[2] = {sep, iter};
+    _Py_IDENTIFIER(join);
+    PyObject *method = _PyUnicode_FromId(&PyId_join); /* borrowed */
+    if (method == NULL) {
+        return NULL;
+    }
+    PyObject *res = PyObject_VectorcallMethod(
```

Comment:
This might fail with 3.8. `PyObject_VectorcallMethod` was provisional there. The leading underscore was removed in `3.9`. `_PyObject_VectorcallMethod` is available as alias in `3.13` for backwards compatibility.

Reply:
This is guarded by CPY_3_13_FEATURES so should be fine

Reply:
True, missed that. Although now `_PyBytes_Join` doesn't need to be fixed for `3.13` anymore. See https://github.com/python/cpython/pull/122287